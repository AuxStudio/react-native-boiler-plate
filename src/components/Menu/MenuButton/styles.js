import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  menuButton: {},
  menuButtonIcon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.primaryTextColor,
  },
});

export default styles;
