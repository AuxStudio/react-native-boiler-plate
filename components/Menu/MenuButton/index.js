import React from 'react';
import PropTypes from 'prop-types';
import { Text, ViewPropTypes } from 'react-native';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  handlePress: PropTypes.func,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {
  iconName: 'menu',
};

const MenuButton = ({ iconName, handlePress, iconStyle, style }) => {
  return (
    <Touchable onPress={handlePress} style={[styles.menuButton, style]}>
      <Icon name={iconName} style={[styles.menuButtonIcon, iconStyle]} />
    </Touchable>
  );
};

MenuButton.propTypes = propTypes;
MenuButton.defaultProps = defaultProps;

export default MenuButton;
