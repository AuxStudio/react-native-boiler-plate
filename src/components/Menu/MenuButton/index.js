import React from 'react';
import PropTypes from 'prop-types';
import { Text, ViewPropTypes } from 'react-native';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  handlePress: PropTypes.func,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
  testID: PropTypes.string,
};

const defaultProps = {
  iconName: 'menu',
};

const MenuButton = ({ iconName, handlePress, iconStyle, style, testID }) => {
  return (
    <Touchable onPress={handlePress} style={[styles.menuButton, style]} testID={testID}>
      <Icon name={iconName} style={[styles.menuButtonIcon, iconStyle]} />
    </Touchable>
  );
};

MenuButton.propTypes = propTypes;
MenuButton.defaultProps = defaultProps;

export default MenuButton;
