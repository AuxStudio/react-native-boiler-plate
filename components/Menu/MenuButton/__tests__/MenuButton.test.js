/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';

import MenuButton from '../';

it('renders a MenuButton', () => {
  expect(renderer.create(<MenuButton handlePress={jest.fn()} />)).toMatchSnapshot();
});

it('renders a MenuButton with minimum required props', () => {
  expect(renderer.create(<MenuButton />)).toMatchSnapshot();
});
/* eslint-enable */
