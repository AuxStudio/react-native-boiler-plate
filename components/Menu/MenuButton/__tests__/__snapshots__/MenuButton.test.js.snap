// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a MenuButton 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`renders a MenuButton with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": undefined,
          "fontSize": 24,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
